Code/GraphMol/RGroupDecomposition/RGroupDecomp.h

```diff
@@ -0,0 +1,136 @@
+//
+//  Copyright (C) 2017 Novartis Institutes for BioMedical Research
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include "../RDKitBase.h"
+#include <GraphMol/Substruct/SubstructMatch.h>
+
+namespace RDKit {
+
+//! Compute the isomporphic degenerative points in the
+//!  molecule.  These are points that are symmetrically
+//!  equivalent.
+/*!
+   \param mol     Molecule to compute the degenerative points
+
+   \return the set of degenerative points (set<unsigned int>)
+*/
+
+typedef enum  {
+  IsotopeLabels          = 0x01,
+  AtomMapLabels          = 0x02,
+  AtomIndexLabels        = 0x04,
+  RelabelDuplicateLabels = 0x08,
+  AutoDetect             = 0x0F,
+} RGroupLabels;
+
+typedef enum  {
+  Greedy        = 0x01,
+  GreedyChunks  = 0x02,
+  Exhaustive    = 0x04, // not really useful for large sets
+} RGroupMatching;
+
+typedef enum  {
+  AtomNum     = 0x01,
+  Isotope     = 0x02,
+  MDLRGroup   = 0x04,
+} RGroupLabelling;
+
+typedef enum  {
+  None    = 0x0,
+  MCS     = 0x01,
+} RGroupCoreAlignment;
+
+struct RGroupDecompositionParameters {
```

Comment:
It does make it a bit easier to expose to python.  I could just make this a class as well.